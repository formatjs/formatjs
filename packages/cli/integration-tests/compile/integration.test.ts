import {exec as nodeExec} from 'child_process'
import {join, resolve} from 'path'
import {promisify} from 'util'
const exec = promisify(nodeExec)
const BIN_PATH = require.resolve('@formatjs/cli/bin/formatjs')
const ARTIFACT_PATH = resolve(__dirname, 'test_artifacts')

test('basic case: help', async () => {
  await expect(exec(`${BIN_PATH} compile --help`)).resolves.toMatchSnapshot()
}, 20000)

test('basic case: empty json', async () => {
  await expect(
    exec(`${BIN_PATH} compile ${join(__dirname, 'lang/empty.json')}`)
  ).resolves.toMatchSnapshot()
}, 20000)

test('normal json', async () => {
  await expect(
    exec(`${BIN_PATH} compile ${join(__dirname, 'lang/en.json')}`)
  ).resolves.toMatchSnapshot()
}, 20000)

test('xx-LS json', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile --ast --pseudo-locale xx-LS ${join(
        __dirname,
        'lang/en.json'
      )}`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('xx-HA json', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile --ast --pseudo-locale xx-HA ${join(
        __dirname,
        'lang/en.json'
      )}`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('xx-AC json', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile --ast --pseudo-locale xx-AC ${join(
        __dirname,
        'lang/en.json'
      )}`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('en-XA json', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile --ast --pseudo-locale en-XA ${join(
        __dirname,
        'lang/en.json'
      )}`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('en-XB json', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile --ast --pseudo-locale en-XB ${join(
        __dirname,
        'lang/en.json'
      )}`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('normal json with formatter', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile ${join(
        __dirname,
        'lang/en-format.json'
      )} --format ${join(__dirname, '../formatter.js')}`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('normal json with transifex', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile ${join(
        __dirname,
        'lang/en-transifex.json'
      )} --format transifex`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('normal json with smartling', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile ${join(
        __dirname,
        'lang/en-smartling.json'
      )} --format smartling`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('normal json with simple', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile ${join(
        __dirname,
        'lang/en-simple.json'
      )} --format simple`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('normal json with lokalise', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile ${join(
        __dirname,
        'lang/en-lokalise.json'
      )} --format lokalise`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('normal json with crowdin', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile ${join(
        __dirname,
        'lang/en-crowdin.json'
      )} --format crowdin`
    )
  ).resolves.toMatchSnapshot()
}, 20000)

test('malformed ICU message json', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile ${join(__dirname, 'lang/malformed-messages.json')}`
    )
  ).rejects.toThrowError('SyntaxError: EXPECT_ARGUMENT_CLOSING_BRACE')
}, 20000)

test('skipped malformed ICU message json', async () => {
  await expect(
    exec(
      `${BIN_PATH} compile  --skip-errors ${join(
        __dirname,
        'lang/malformed-messages.json'
      )}`
    )
  ).resolves.toMatchSnapshot({
    stderr: expect.stringMatching(
      /^\[@formatjs\/cli\] \[WARN\] Error validating message "my name is {name" with ID "a1dd2" in file .*\/packages\/cli\/integration-tests\/compile\/lang\/malformed-messages.json/
    ),
  } as Partial<any>)
}, 20000)

test('AST', async () => {
  await expect(
    exec(`${BIN_PATH} compile --ast ${join(__dirname, 'lang/en.json')}`)
  ).resolves.toMatchSnapshot()
}, 20000)

test('out-file', async () => {
  const outFilePath = join(ARTIFACT_PATH, 'en.json')
  await expect(
    exec(
      `${BIN_PATH} compile ${join(
        __dirname,
        'lang/en.json'
      )} --out-file ${outFilePath}`
    )
  ).resolves.toMatchSnapshot()
  expect(require(outFilePath)).toMatchSnapshot()
}, 20000)

test('out-file --ast', async () => {
  const outFilePath = join(ARTIFACT_PATH, 'ast.json')
  await expect(
    exec(
      `${BIN_PATH} compile --ast ${join(
        __dirname,
        'lang/en.json'
      )} --out-file ${outFilePath}`
    )
  ).resolves.toMatchSnapshot()
  expect(require(outFilePath)).toMatchSnapshot()
}, 20000)

test('compile glob', async () => {
  await expect(
    exec(`${BIN_PATH} compile "${join(__dirname, 'glob/*.json')}"`)
  ).resolves.toMatchSnapshot()
})

test('compile glob with conflict', async () => {
  await expect(
    exec(`${BIN_PATH} compile "${join(__dirname, 'glob-conflict/*.json')}"`)
  ).rejects.toThrowError(
    'Conflicting ID "a1d12" with different translation found in these 2 files'
  )
})
