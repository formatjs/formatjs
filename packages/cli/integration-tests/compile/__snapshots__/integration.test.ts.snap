// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AST 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": [
    {
      "type": 0,
      "value": "I have "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "type": 0,
              "value": "a dog"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "a1d13": [
    {
      "type": 0,
      "value": "I have "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "a dog"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "a1d14": [
    {
      "type": 0,
      "value": "I have "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "a "
                },
                {
                  "children": [
                    {
                      "type": 0,
                      "value": "dog"
                    }
                  ],
                  "type": 8,
                  "value": "i"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "a1dd2": [
    {
      "type": 0,
      "value": "my name is "
    },
    {
      "type": 1,
      "value": "name"
    }
  ],
  "ashd2": [
    {
      "type": 0,
      "value": "a message"
    }
  ]
}
",
}
`;

exports[`basic case: empty json 1`] = `
{
  "stderr": "",
  "stdout": "{
}
",
}
`;

exports[`basic case: help 1`] = `
{
  "stderr": "",
  "stdout": "Usage: formatjs compile [options] [translation_files...]

Compile extracted translation file into react-intl consumable JSON
We also verify that the messages are valid ICU and not malformed. 
<translation_files> can be a glob like "foo/**/en.json"

Options:
  --format <path>                 Path to a formatter file that converts \`<translation_file>\` to \`Record<string, string>\`
  so we can compile. The file must export a function named \`compile\` with the signature:
  \`\`\`
  type CompileFn = <T = Record<string, MessageDescriptor>>(
    msgs: T
  ) => Record<string, string>;
  \`\`\`
  This is especially useful to convert from a TMS-specific format back to react-intl format
  
  --out-file <path>               Compiled translation output file.
                                  If this is not provided, result will be
                                  printed to stdout
  --ast                           Whether to compile to AST. See
                                  https://formatjs.io/docs/guides/advanced-usage#pre-parsing-messages
                                  for more information
  --skip-errors                   Whether to continue compiling messages after
                                  encountering an error. Any keys with errors
                                  will not be included in the output file.
  --pseudo-locale <pseudoLocale>  Whether to generate pseudo-locale files. See
                                  https://formatjs.io/docs/tooling/cli#--pseudo-locale-pseudolocale
                                  for possible values.
                                  "--ast" is required for this to work.
  -h, --help                      display help for command
",
}
`;

exports[`compile glob 1`] = `
{
  "stderr": "",
  "stdout": "{
  "1": "a message",
  "2": "my name is {name}",
  "3": "I have {count, plural, one{a dog} other{many dogs}}",
  "a1d12": "I have {count, plural, one{a dog} other{many dogs}}",
  "a1dd2": "my name is {name}",
  "ashd2": "a message"
}
",
}
`;

exports[`en-XA json 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": [
    {
      "type": 0,
      "value": "["
    },
    {
      "type": 0,
      "value": "Ī ħȧȧṽḗḗ "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "type": 0,
              "value": "ȧȧ ḓǿǿɠ"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "ḿȧȧƞẏ ḓǿǿɠş"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    },
    {
      "type": 0,
      "value": "]"
    }
  ],
  "a1d13": [
    {
      "type": 0,
      "value": "["
    },
    {
      "type": 0,
      "value": "Ī ħȧȧṽḗḗ "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "ȧȧ ḓǿǿɠ"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "ḿȧȧƞẏ ḓǿǿɠş"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    },
    {
      "type": 0,
      "value": "]"
    }
  ],
  "a1d14": [
    {
      "type": 0,
      "value": "["
    },
    {
      "type": 0,
      "value": "Ī ħȧȧṽḗḗ "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "ȧȧ "
                },
                {
                  "children": [
                    {
                      "type": 0,
                      "value": "ḓǿǿɠ"
                    }
                  ],
                  "type": 8,
                  "value": "i"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "ḿȧȧƞẏ ḓǿǿɠş"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    },
    {
      "type": 0,
      "value": "]"
    }
  ],
  "a1dd2": [
    {
      "type": 0,
      "value": "["
    },
    {
      "type": 0,
      "value": "ḿẏ ƞȧȧḿḗḗ īş "
    },
    {
      "type": 1,
      "value": "name"
    },
    {
      "type": 0,
      "value": "]"
    }
  ],
  "ashd2": [
    {
      "type": 0,
      "value": "["
    },
    {
      "type": 0,
      "value": "ȧȧ ḿḗḗşşȧȧɠḗḗ"
    },
    {
      "type": 0,
      "value": "]"
    }
  ]
}
",
}
`;

exports[`en-XB json 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": [
    {
      "type": 0,
      "value": "‮"
    },
    {
      "type": 0,
      "value": "I ɥɐʌǝ "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "type": 0,
              "value": "ɐ poƃ"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "ɯɐuʎ poƃs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    },
    {
      "type": 0,
      "value": "‬"
    }
  ],
  "a1d13": [
    {
      "type": 0,
      "value": "‮"
    },
    {
      "type": 0,
      "value": "I ɥɐʌǝ "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "ɐ poƃ"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "ɯɐuʎ poƃs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    },
    {
      "type": 0,
      "value": "‬"
    }
  ],
  "a1d14": [
    {
      "type": 0,
      "value": "‮"
    },
    {
      "type": 0,
      "value": "I ɥɐʌǝ "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "ɐ "
                },
                {
                  "children": [
                    {
                      "type": 0,
                      "value": "poƃ"
                    }
                  ],
                  "type": 8,
                  "value": "i"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "ɯɐuʎ poƃs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    },
    {
      "type": 0,
      "value": "‬"
    }
  ],
  "a1dd2": [
    {
      "type": 0,
      "value": "‮"
    },
    {
      "type": 0,
      "value": "ɯʎ uɐɯǝ ıs "
    },
    {
      "type": 1,
      "value": "name"
    },
    {
      "type": 0,
      "value": "‬"
    }
  ],
  "ashd2": [
    {
      "type": 0,
      "value": "‮"
    },
    {
      "type": 0,
      "value": "ɐ ɯǝssɐƃǝ"
    },
    {
      "type": 0,
      "value": "‬"
    }
  ]
}
",
}
`;

exports[`normal json 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": "I have {count, plural, one{a dog} other{many dogs}}",
  "a1d13": "I have {count, plural, one{<b>a dog</b>} other{many dogs}}",
  "a1d14": "I have {count, plural, one{<b>a <i>dog</i></b>} other{many dogs}}",
  "a1dd2": "my name is {name}",
  "ashd2": "a message"
}
",
}
`;

exports[`normal json with crowdin 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": "I have {count, plural, one{a dog} other{many dogs}}",
  "a1dd2": "my name is {name}",
  "ashd2": "a message"
}
",
}
`;

exports[`normal json with formatter 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": "I have {count, plural, one{a dog} other{many dogs}}",
  "a1dd2": "my name is {name}",
  "ashd2": "a message"
}
",
}
`;

exports[`normal json with lokalise 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": "I have {count, plural, one{a dog} other{many dogs}}",
  "a1dd2": "my name is {name}",
  "ashd2": "a message"
}
",
}
`;

exports[`normal json with simple 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": "I have {count, plural, one{a dog} other{many dogs}}",
  "a1dd2": "my name is {name}",
  "ashd2": "a message"
}
",
}
`;

exports[`normal json with smartling 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": "I have {count, plural, one{a dog} other{many dogs}}",
  "a1dd2": "my name is {name}",
  "ashd2": "a message"
}
",
}
`;

exports[`normal json with transifex 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": "I have {count, plural, one{a dog} other{many dogs}}",
  "a1dd2": "my name is {name}",
  "ashd2": "a message"
}
",
}
`;

exports[`out-file --ast 1`] = `
{
  "stderr": "",
  "stdout": "",
}
`;

exports[`out-file --ast 2`] = `
{
  "a1d12": [
    {
      "type": 0,
      "value": "I have ",
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "type": 0,
              "value": "a dog",
            },
          ],
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs",
            },
          ],
        },
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count",
    },
  ],
  "a1d13": [
    {
      "type": 0,
      "value": "I have ",
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "a dog",
                },
              ],
              "type": 8,
              "value": "b",
            },
          ],
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs",
            },
          ],
        },
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count",
    },
  ],
  "a1d14": [
    {
      "type": 0,
      "value": "I have ",
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "a ",
                },
                {
                  "children": [
                    {
                      "type": 0,
                      "value": "dog",
                    },
                  ],
                  "type": 8,
                  "value": "i",
                },
              ],
              "type": 8,
              "value": "b",
            },
          ],
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs",
            },
          ],
        },
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count",
    },
  ],
  "a1dd2": [
    {
      "type": 0,
      "value": "my name is ",
    },
    {
      "type": 1,
      "value": "name",
    },
  ],
  "ashd2": [
    {
      "type": 0,
      "value": "a message",
    },
  ],
}
`;

exports[`out-file 1`] = `
{
  "stderr": "",
  "stdout": "",
}
`;

exports[`out-file 2`] = `
{
  "a1d12": "I have {count, plural, one{a dog} other{many dogs}}",
  "a1d13": "I have {count, plural, one{<b>a dog</b>} other{many dogs}}",
  "a1d14": "I have {count, plural, one{<b>a <i>dog</i></b>} other{many dogs}}",
  "a1dd2": "my name is {name}",
  "ashd2": "a message",
}
`;

exports[`skipped malformed ICU message json 1`] = `
{
  "stderr": StringMatching /\\^\\\\\\[@formatjs\\\\/cli\\\\\\] \\\\\\[WARN\\\\\\] Error validating message "my name is \\{name" with ID "a1dd2" in file \\.\\*\\\\/packages\\\\/cli\\\\/integration-tests\\\\/compile\\\\/lang\\\\/malformed-messages\\.json/,
  "stdout": "{
  "a1d12": "I have {count, plural, one{a dog} other{many dogs}}",
  "ashd2": "a message"
}
",
}
`;

exports[`xx-AC json 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": [
    {
      "type": 0,
      "value": "I HAVE "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "type": 0,
              "value": "A DOG"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "MANY DOGS"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "a1d13": [
    {
      "type": 0,
      "value": "I HAVE "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "A DOG"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "MANY DOGS"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "a1d14": [
    {
      "type": 0,
      "value": "I HAVE "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "A "
                },
                {
                  "children": [
                    {
                      "type": 0,
                      "value": "DOG"
                    }
                  ],
                  "type": 8,
                  "value": "i"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "MANY DOGS"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "a1dd2": [
    {
      "type": 0,
      "value": "MY NAME IS "
    },
    {
      "type": 1,
      "value": "name"
    }
  ],
  "ashd2": [
    {
      "type": 0,
      "value": "A MESSAGE"
    }
  ]
}
",
}
`;

exports[`xx-HA json 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": [
    {
      "type": 0,
      "value": "[javascript]I have "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "type": 0,
              "value": "a dog"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "a1d13": [
    {
      "type": 0,
      "value": "[javascript]I have "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "a dog"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "a1d14": [
    {
      "type": 0,
      "value": "[javascript]I have "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "a "
                },
                {
                  "children": [
                    {
                      "type": 0,
                      "value": "dog"
                    }
                  ],
                  "type": 8,
                  "value": "i"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "a1dd2": [
    {
      "type": 0,
      "value": "[javascript]my name is "
    },
    {
      "type": 1,
      "value": "name"
    }
  ],
  "ashd2": [
    {
      "type": 0,
      "value": "[javascript]a message"
    }
  ]
}
",
}
`;

exports[`xx-LS json 1`] = `
{
  "stderr": "",
  "stdout": "{
  "a1d12": [
    {
      "type": 0,
      "value": "I have "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "type": 0,
              "value": "a dog"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    },
    {
      "type": 0,
      "value": "SSSSSSSSSSSSSSSSSSSSSSSSS"
    }
  ],
  "a1d13": [
    {
      "type": 0,
      "value": "I have "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "a dog"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    },
    {
      "type": 0,
      "value": "SSSSSSSSSSSSSSSSSSSSSSSSS"
    }
  ],
  "a1d14": [
    {
      "type": 0,
      "value": "I have "
    },
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "children": [
                {
                  "type": 0,
                  "value": "a "
                },
                {
                  "children": [
                    {
                      "type": 0,
                      "value": "dog"
                    }
                  ],
                  "type": 8,
                  "value": "i"
                }
              ],
              "type": 8,
              "value": "b"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 0,
              "value": "many dogs"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    },
    {
      "type": 0,
      "value": "SSSSSSSSSSSSSSSSSSSSSSSSS"
    }
  ],
  "a1dd2": [
    {
      "type": 0,
      "value": "my name is "
    },
    {
      "type": 1,
      "value": "name"
    },
    {
      "type": 0,
      "value": "SSSSSSSSSSSSSSSSSSSSSSSSS"
    }
  ],
  "ashd2": [
    {
      "type": 0,
      "value": "a messageSSSSSSSSSSSSSSSSSSSSSSSSS"
    }
  ]
}
",
}
`;
